compiler/src/linking/link.re

```diff
@@ -4,24 +4,9 @@ open Grain_typed;
 open Grain_utils;
 open Cmi_format;
 open Binaryen;
-open Graph;
-
-module String = {
-  type t = string;
-  let compare = compare;
-  let hash = Hashtbl.hash;
-  let equal = (==);
-  let default = "";
-};
-
-module G = Imperative.Digraph.Concrete(String);
-module Topo = Topological.Make(G);
 
-let dependency_graph = G.create(~size=10, ());
 let modules: Hashtbl.t(string, Module.t) = Hashtbl.create(10);
-
-let grain_main = "_gmain";
-let grain_start = "_start";
+let hard_dependencies: Hashtbl.t(string, unit) = Hashtbl.create(10);
```

Comment:
Hard dependencies are ones that require code to be loaded from the module, whereas soft dependencies just require type information be loaded.